### PR TITLE
Extended AutoSuggestion SingleLineEditor to multiline 

### PR DESCRIPTION
**Pull request by arausly**

### CODE REVIEW

File: src/components/AutoSuggestion/AutoSuggestion.tsx [hunk 1]
```diff
@@ -471,6 +496,14 @@ export const AutoSuggestion = ({
                 default:
                 //do nothing
             }
+        } else {
+            if (multiline && editorState.editorInstance && keyPressedFromInput === OVERWRITTEN_KEYS.Enter) {
```
Comment (andreas-schultz) [inline src/components/AutoSuggestion/AutoSuggestion.tsx:500]: This does not seem to work reliably, e.g. by enter random chars quickly followed by pressing the Enter key does not always create a newline. 
Shouldn't this be handled in `handleInputEditorKeyPress` instead by not swallowing the ENTER key press?
There seems to be a UX conflict here. The arrow and Enter keys have other semantics in the multiline case compared to the single line editor case.

File: src/components/AutoSuggestion/SingleLineCodeEditor.tsx [hunk 2]
```diff
@@ -1,38 +1,39 @@
...
+import { CLASSPREFIX as eccgui } from "../../configuration/constants";
 
 export interface IRange {
-  from: number
-  to: number
+    from: number;
+    to: number;
 }
 
 export interface SingleLineCodeEditorProps {
```
Comment (andreas-schultz) [inline src/components/AutoSuggestion/SingleLineCodeEditor.tsx:14]: The component should be renamed to reflect the multi-line option, e.g. `CodeEditor`.

File: src/components/AutoSuggestion/SingleLineCodeEditor.tsx [hunk 3]
```diff
@@ -1,38 +1,39 @@
...
+    onSelection: (ranges: IRange[]) => any;
     // If the <Tab> key is enabled as normal input, i.e. it won't have the behavior of changing to the next input element, expected in a web app.
-    enableTab?: boolean
+    enableTab?: boolean;
     /** Placeholder tobe shown when no text has been entered, yet. */
-    placeholder?: string
-    showScrollBar?: boolean
+    placeholder?: string;
+    showScrollBar?: boolean;
+    multiline?: boolean;
```
Comment (andreas-schultz) [inline src/components/AutoSuggestion/SingleLineCodeEditor.tsx:36]: Missing parameter documentation.

**Review by andreas-schultz — APPROVED** [review]

File: src/components/AutoSuggestion/SingleLineCodeEditor.tsx [hunk 4]
```diff
@@ -1,110 +0,0 @@
...
-    enableTab?: boolean
-    /** Placeholder tobe shown when no text has been entered, yet. */
-    placeholder?: string
-    showScrollBar?: boolean
-}
-
-export type IEditorProps = SingleLineCodeEditorProps;
-
-/** A single-line code editor. */
-export const SingleLineCodeEditor = ({
```
Comment (haschek) [inline src/components/AutoSuggestion/SingleLineCodeEditor.tsx:41]: @arausly @andreas-schultz This element should not have been removed as it was exported. Pls just mark it as deprecated next time with a short note what element should be now used. Removing exported elements leads to breaking changes.

File: src/extensions/codemirror/_codemirror.scss [hunk 5]
```diff
@@ -9,6 +9,7 @@
     .CodeMirror {
         clip-path: unset !important; // we may check later why they set inset(0) now
         border-radius: $pt-border-radius;
+        height: 290px;
```
Comment (haschek) [inline src/extensions/codemirror/_codemirror.scss:12]: @arausly Looks like a forgotten attempt to fix the flipping issue, correct?

**Pull request merged**